connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -527,7 +531,9 @@ public Void call() throws Exception {
 
                         // Note that we use the updated connector config despite the fact that we don't have an updated
                         // snapshot yet. The existing task info should still be accurate.
-                        ConnectorInfo info = new ConnectorInfo(connName, config, configState.tasks(connName));
+                        Map<String, String> map = configState.connectorConfig(connName);
+                        ConnectorInfo info = new ConnectorInfo(connName, config, configState.tasks(connName),
+                            map == null ? null : WorkerConnector.getConnectorType(map.get(ConnectorConfig.CONNECTOR_CLASS_CONFIG)));
```

Comment:
Same thing here as above: probably need to use `worker.getConnectorType(className)` here.